crates/core/tedge_agent/src/agent.rs

```diff
@@ -275,13 +301,18 @@ impl SmAgent {
 
         self.process_pending_operation(&mut mqtt.published).await?;
 
-        while let Err(error) = self
-            .process_subscribed_messages(&mut mqtt.received, &mut mqtt.published, &plugins)
-            .await
-        {
-            error!("{}", error);
-        }
+        // TODO: how to break out? the agent should never break out? right?
+        // NOTE: a loop is not required here
+        tokio::select! {
+            Err(error) = self
+            .process_subscribed_messages(&mut mqtt.received, &mut mqtt.published, &plugins) => {
+                error!("{}", error);
+            }
+            Err(error) = server => {
+                error!("{}", error);
+            }
 
+        };
```

Comment:
This usage of `select!` sounds strange to me. Why not simply spawn the http server in the background?